index.js

```diff
@@ -44,9 +64,26 @@ function mongo(options) {
     log : log 
   });
 
+  let mongoPool = createMongoPool();
+
   return function* koaMongo(next) {
+
     this.mongo = yield mongoPool.acquire.bind(mongoPool);
+
     if (!this.mongo) this.throw('Fail to acquire one mongo connection');
+
+    if (!(this.mongo).serverConfig.isConnected()) {
```

Comment:
IMO:

```js
if (!(this.mongo).serverConfig.isConnected()) {
  mongoPool.release(this.mongo)
  this.throw('Cannot connect to MongoDB')
}
```
Throw the connection error directly rather than waiting MongoDB restart.